### PR TITLE
[Feature] support torchserve

### PR DESCRIPTION
**Pull request by nijkah**
## Motivation
closes https://github.com/open-mmlab/mmrotate/issues/38
Support to deploy to torchserve

## Modification
add tools and sample Dockerfile

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. The documentation has been modified accordingly, like docstring or example tutorials.

- [x] test `tools/deployment/mmrotate2serve.py` with **configs/rotated_faster_rcnn/rotated_faster_rcnn_r50_fpn_1x_dota_le90.py**
- [x] test Dockerfile and torchserve
- [x] add a documentation

### CODE REVIEW

**Comment by zytx121** [comment]
Good job! Thanks for your PR.

**Comment by nijkah** [comment]
Note that
1. I added brief document for torchserve in English. But I cannot speak Chinese. Any help would be appreciated.
2. There is a version issue in torchserve. See [this issue](https://github.com/pytorch/serve/issues/1548). It should be fixed in mmdetection too. There are three options. I followed the second one.
- 1. Upgrade the version of torch>=1.8.1
- 2. Limit the version of torchserve==0.2.0
- 3. Wait for the latest torchserve to fix the issue.

3. There is a segmentation postprocessing part in `tools/deployment/mmrotate_handler.py`. Should I delete this?

**Review by liuyanyi — APPROVED** [review]
LGTM

File: tools/deployment/mmrotate_handler.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+from mmdet.apis import inference_detector, init_detector
+from ts.torch_handler.base_handler import BaseHandler
+
+import mmrotate  # noqa: F401
+
+
+class MMRotateHandler(BaseHandler):
+    threshold = 0.5
+
+    def initialize(self, context):
```
Comment (yangxue0827) [inline tools/deployment/mmrotate_handler.py:16]: Docstring is required, like-wise for `preprocess`, `inference` and `postprocess`.

File: docs/en/useful_tools.md [hunk 2]
```diff
@@ -60,6 +60,132 @@ Examples:
...
+    "bbox": [
+      152.0239715576172,
+      305.92572021484375,
+      43.144744873046875,
+      18.85024642944336,
+      0.014928221702575684
+    ],
+    "score": 0.972826361656189
+  },
+  # ...
```
Comment (yangxue0827) [inline docs/en/useful_tools.md:146]: I'm not sure if this is normal.

Comment (nijkah) [reply docs/en/useful_tools.md:146]: Because it is a just sample of json, I can delete it.

**Review by yangxue0827 — APPROVED** [review]
LGTM

**Review by zytx121 — APPROVED** [review]
LGTM

**Pull request merged**